### PR TITLE
[Proposal] Move RequestIdentifierFeature to HttpContext

### LINKED ISSUE #412: Expose TraceIdentifier as a property on HttpContext
It's pretty convenient to make available.

### PR DESCRIPTION
**Pull request by benaadams**
Allow lazier instantiation
Expose TraceIdentifier on Httpcontext
Also resolves #412

Add tests

### CODE REVIEW

File: src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNet.Http.Features.Internal
+{
+    public class DefaultHttpRequestIdentifierFeature : HttpRequestIdentifierFeature, IHttpRequestIdentifierFeature
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs]: There's no reason for DefaultHttpRequestIdentifierFeature to exist independent of HttpRequestIdentifierFeature, just merge them.

File: src/Microsoft.AspNet.Http/project.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
         "url": "git://github.com/aspnet/httpabstractions"
     },
     "compilationOptions": {
-        "warningsAsErrors": true
+			"warningsAsErrors": true,
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http/project.json]: nit: spacing

File: src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs]: Yar

File: src/Microsoft.AspNet.Http/project.json (hunk 2 above)
Comment (benaadams) [inline src/Microsoft.AspNet.Http/project.json]: Done

File: src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs (hunk 1 above)
Comment (benaadams) [inline src/Microsoft.AspNet.Http/DefaultHttpRequestIdentifierFeature.cs]: Done

**Comment by davidfowl** [comment]
@benaadams can you rebase this?

**Comment by benaadams** [comment]
Rebased (rather recreated)  in #440

**Pull request closed without merging**